onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -407,9 +432,9 @@ Status CUDAExecutionProvider::OnRunEnd(bool sync_stream) {
       GetPerThreadContext().IncrementRegularRunCountBeforeGraphCapture();
     }
   }
-  // record deferred release event on default stream, and release per_thread_context
-  auto current_deferred_release_event = GetPerThreadContext().GetCurrentDeferredReleaseEvent();
-  CUDA_RETURN_IF_ERROR(cudaEventRecord(current_deferred_release_event, static_cast<cudaStream_t>(GetComputeStream())));
+
+  EnqueueDeferredRelease();
```

Comment:
actually why we want to enqueue it at OnRunStart? i thought enqueue it at OnRunEnd seems more natural.

Reply:
It's OnRunEnd.
![image](https://user-images.githubusercontent.com/3524474/188991829-b3b14dc0-ba01-4f22-ad33-647d78a759cd.png)